[
  {
    "timestamp": "2024-10-25 00:53:32",
    "date": "2024-10-25",
    "time": "00:53:32",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:53:32",
    "date": "2024-10-25",
    "time": "00:53:32",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:53:32",
    "date": "2024-10-25",
    "time": "00:53:32",
    "symbol": "BTCUSDT",
    "message": "ไม่พบไฟล์สถานะ เริ่มต้นด้วยค่าเริ่มต้น (Timeframe: 1h)",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 00:55:11",
    "date": "2024-10-25",
    "time": "00:55:11",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:55:11",
    "date": "2024-10-25",
    "time": "00:55:11",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:55:11",
    "date": "2024-10-25",
    "time": "00:55:11",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:58:20",
    "date": "2024-10-25",
    "time": "00:58:20",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:58:21",
    "date": "2024-10-25",
    "time": "00:58:21",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 00:58:21",
    "date": "2024-10-25",
    "time": "00:58:21",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 01:03:30",
    "date": "2024-10-25",
    "time": "01:03:30",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 01:03:30",
    "date": "2024-10-25",
    "time": "01:03:30",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 01:03:30",
    "date": "2024-10-25",
    "time": "01:03:30",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 01:06:00",
    "date": "2024-10-25",
    "time": "01:06:00",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 560, in _request\n    await resp.start(conn)\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client_reqrep.py\", line 894, in start\n    with self._timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://dapi.binance.com/dapi/v1/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 01:06:03",
    "date": "2024-10-25",
    "time": "01:06:03",
    "symbol": "BTCUSDT",
    "message": "ไม่สามารถดึงราคาตลาดได้ ข้ามรอบนี้",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 02:03:51",
    "date": "2024-10-25",
    "time": "02:03:51",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 560, in _request\n    await resp.start(conn)\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client_reqrep.py\", line 894, in start\n    with self._timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://api.binance.com/api/v3/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 02:03:52",
    "date": "2024-10-25",
    "time": "02:03:52",
    "symbol": "BTCUSDT",
    "message": "ไม่สามารถดึงราคาตลาดได้ ข้ามรอบนี้",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 03:01:10",
    "date": "2024-10-25",
    "time": "03:01:10",
    "symbol": "BTCUSDT",
    "message": "ตั้งคำสั่ง BUY STOP_MARKET ที่ราคา 68302.82960000",
    "color": "blue"
  },
  {
    "timestamp": "2024-10-25 03:01:10",
    "date": "2024-10-25",
    "time": "03:01:10",
    "symbol": "BTCUSDT",
    "message": "ตั้งคำสั่ง Stoploss ที่ราคา 67693.23000000",
    "color": "blue"
  },
  {
    "timestamp": "2024-10-25 03:28:27",
    "date": "2024-10-25",
    "time": "03:28:27",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 467, in _request\n    with timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://fapi.binance.com/fapi/v1/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 03:57:58",
    "date": "2024-10-25",
    "time": "03:57:58",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 560, in _request\n    await resp.start(conn)\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client_reqrep.py\", line 894, in start\n    with self._timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://dapi.binance.com/dapi/v1/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 03:58:01",
    "date": "2024-10-25",
    "time": "03:58:01",
    "symbol": "BTCUSDT",
    "message": "ไม่สามารถดึงราคาตลาดได้ ข้ามรอบนี้",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 04:08:09",
    "date": "2024-10-25",
    "time": "04:08:09",
    "symbol": "BTCUSDT",
    "message": "Entry order ทำงานที่ราคา 68306.20000000",
    "color": "green"
  },
  {
    "timestamp": "2024-10-25 04:09:56",
    "date": "2024-10-25",
    "time": "04:09:56",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 560, in _request\n    await resp.start(conn)\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client_reqrep.py\", line 894, in start\n    with self._timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://fapi.binance.com/fapi/v1/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 04:09:59",
    "date": "2024-10-25",
    "time": "04:09:59",
    "symbol": "BTCUSDT",
    "message": "ไม่สามารถดึงราคาตลาดได้ ข้ามรอบนี้",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 05:00:21",
    "date": "2024-10-25",
    "time": "05:00:21",
    "symbol": "BTCUSDT",
    "message": "กำลังพิจารณาแท่งเทียนที่ปิดแล้ว 5 แท่ง - Low prices: 67436.60, 67603.80, 67700.00, 67890.00, 68172.60",
    "color": "blue"
  },
  {
    "timestamp": "2024-10-25 05:00:21",
    "date": "2024-10-25",
    "time": "05:00:21",
    "symbol": "BTCUSDT",
    "message": "เวลาของแท่งเทียนที่พิจารณา: 17:00, 18:00, 19:00, 20:00, 21:00",
    "color": "blue"
  },
  {
    "timestamp": "2024-10-25 05:00:29",
    "date": "2024-10-25",
    "time": "05:00:29",
    "symbol": "BTCUSDT",
    "message": "สร้างคำสั่ง stop loss ใหม่ที่ราคา 67693.23",
    "color": "white"
  },
  {
    "timestamp": "2024-10-25 05:00:31",
    "date": "2024-10-25",
    "time": "05:00:31",
    "symbol": "BTCUSDT",
    "message": "ปรับ stoploss จาก 67693.20 เป็น 67693.23 (พิจารณาจากแท่งที่ปิดแล้ว: 67700.00, 67890.00, 68172.60)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 09:41:04",
    "date": "2024-10-25",
    "time": "09:41:04",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 09:41:04",
    "date": "2024-10-25",
    "time": "09:41:04",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 09:41:04",
    "date": "2024-10-25",
    "time": "09:41:04",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 09:43:06",
    "date": "2024-10-25",
    "time": "09:43:06",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 560, in _request\n    await resp.start(conn)\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client_reqrep.py\", line 894, in start\n    with self._timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://dapi.binance.com/dapi/v1/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 09:43:08",
    "date": "2024-10-25",
    "time": "09:43:08",
    "symbol": "BTCUSDT",
    "message": "ไม่สามารถดึงราคาตลาดได้ ข้ามรอบนี้",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 09:53:15",
    "date": "2024-10-25",
    "time": "09:53:15",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 09:53:15",
    "date": "2024-10-25",
    "time": "09:53:15",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 09:53:15",
    "date": "2024-10-25",
    "time": "09:53:15",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:00:46",
    "date": "2024-10-25",
    "time": "10:00:46",
    "symbol": "BTCUSDT",
    "message": "กำลังพิจารณาแท่งเทียนที่ปิดแล้ว 5 แท่ง - Low prices: 68158.30, 68030.30, 67883.40, 67816.70, 67800.00",
    "color": "blue"
  },
  {
    "timestamp": "2024-10-25 10:00:46",
    "date": "2024-10-25",
    "time": "10:00:46",
    "symbol": "BTCUSDT",
    "message": "เวลาของแท่งเทียนที่พิจารณา: 22:00, 23:00, 00:00, 01:00, 02:00",
    "color": "blue"
  },
  {
    "timestamp": "2024-10-25 10:00:46",
    "date": "2024-10-25",
    "time": "10:00:46",
    "symbol": "BTCUSDT",
    "message": "ไม่พบชุดแท่งเทียนที่เข้าเงื่อนไข",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 10:05:35",
    "date": "2024-10-25",
    "time": "10:05:35",
    "symbol": "BTCUSDT",
    "message": "Error: Traceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 199, in fetch\n    async with session_method(yarl.URL(url, encoded=True),\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 1141, in __aenter__\n    self._resp = await self._coro\n                 ^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client.py\", line 560, in _request\n    await resp.start(conn)\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\client_reqrep.py\", line 894, in start\n    with self._timer:\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\aiohttp\\helpers.py\", line 721, in __exit__\n    raise asyncio.TimeoutError from None\nTimeoutError\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"C:\\Users\\Chattrawut_P\\Desktop\\coin_Script\\function\\binance\\futures\\order\\other\\get_future_market_price.py\", line 18, in get_future_market_price\n    ticker = await exchange.fetch_ticker(symbol)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 276, in load_markets\n    raise e\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 272, in load_markets\n    result = await self.markets_loading\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 262, in load_markets_helper\n    markets = await self.fetch_markets(params)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 2714, in fetch_markets\n    promises = await asyncio.gather(*promisesRaw)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\binance.py\", line 9406, in request\n    response = await self.fetch2(path, api, method, params, headers, body, config)\n               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 781, in fetch2\n    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Chattrawut_P\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\ccxt\\async_support\\base\\exchange.py\", line 233, in fetch\n    raise RequestTimeout(details) from e\nccxt.base.errors.RequestTimeout: binance GET https://api.binance.com/api/v3/exchangeInfo\n",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 10:05:35",
    "date": "2024-10-25",
    "time": "10:05:35",
    "symbol": "BTCUSDT",
    "message": "ไม่สามารถดึงราคาตลาดได้",
    "color": "red"
  },
  {
    "timestamp": "2024-10-25 10:22:00",
    "date": "2024-10-25",
    "time": "10:22:00",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:22:03",
    "date": "2024-10-25",
    "time": "10:22:03",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:22:03",
    "date": "2024-10-25",
    "time": "10:22:03",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:35:50",
    "date": "2024-10-25",
    "time": "10:35:50",
    "symbol": "BTCUSDT",
    "message": "สร้าง tasks สำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:35:53",
    "date": "2024-10-25",
    "time": "10:35:53",
    "symbol": "BTCUSDT",
    "message": "เริ่มต้นบอทสำหรับ BTCUSDT (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:35:53",
    "date": "2024-10-25",
    "time": "10:35:53",
    "symbol": "BTCUSDT",
    "message": "System Startup - โหลดสถานะเสร็จสมบูรณ์ (Timeframe: 1h)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:59:32",
    "date": "2024-10-25",
    "time": "10:59:32",
    "symbol": "BTCUSDT",
    "message": "ไม่พบ Position ที่เปิดอยู่",
    "color": "yellow"
  },
  {
    "timestamp": "2024-10-25 10:59:44",
    "date": "2024-10-25",
    "time": "10:59:44",
    "symbol": "BTCUSDT",
    "message": "บันทึกการเทรด: BUY BTCUSDT",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:59:44",
    "date": "2024-10-25",
    "time": "10:59:44",
    "symbol": "BTCUSDT",
    "message": "Entry: 68306.20 | Exit: 67958.40",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:59:44",
    "date": "2024-10-25",
    "time": "10:59:44",
    "symbol": "BTCUSDT",
    "message": "จำนวน: 0.00200000 (136.61 USD)",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:59:44",
    "date": "2024-10-25",
    "time": "10:59:44",
    "symbol": "BTCUSDT",
    "message": "Leverage: 20x | Margin Type: cross",
    "color": "cyan"
  },
  {
    "timestamp": "2024-10-25 10:59:44",
    "date": "2024-10-25",
    "time": "10:59:44",
    "symbol": "BTCUSDT",
    "message": "กำไร/ขาดทุน: -0.70 USDT (-0.51%)",
    "color": "cyan"
  }
]